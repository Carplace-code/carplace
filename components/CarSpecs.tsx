/* eslint-disable */

export default function CarSpecs({ listing }: { listing: any }) {
  const { trim } = listing;
  const { version } = trim;
  const { model } = version;
  const { brand } = model;

  return (
    <div className="space-y-3 rounded-md border bg-white p-6">
      <h2 className="mb-4 text-lg font-semibold">Vehicle Specifications</h2>
      <div className="grid grid-cols-2 gap-3 text-sm">
        <p>
          <strong>Brand/Model:</strong> {brand.name} {model.name}
        </p>
        <p>
          <strong>Color:</strong> {listing.exteriorColor}
        </p>
        <p>
          <strong>Year:</strong> {version.year}
        </p>
        <p>
          <strong>Engine:</strong> {trim.motorSize} cc
        </p>
        <p>
          <strong>Transmission:</strong> {trim.transmissionType}
        </p>
        <p>
          <strong>VIN:</strong> N/A
        </p>
        <p>
          <strong>Fuel Type:</strong> {trim.fuelType}
        </p>
      </div>
      <button className="mt-4 rounded-md bg-blue-600 px-4 py-2 text-sm text-white hover:bg-blue-700" type="button">
        Check Vehicle History
      </button>
    </div>
  );
}
