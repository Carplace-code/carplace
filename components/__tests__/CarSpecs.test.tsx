import { render } from "@testing-library/react";
import { describe, it } from "vitest";

import CarSpecs from "../CarSpecs";

describe("CarSpecs", () => {
  it("renders correctly and matches snapshot", () => {
    const { container } = render(<CarSpecs />);
    expect(container).toMatchSnapshot();
  });
});
