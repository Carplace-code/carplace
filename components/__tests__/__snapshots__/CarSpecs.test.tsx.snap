// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CarSpecs > renders correctly and matches snapshot 1`] = `
<div>
  <div
    class="space-y-3 rounded-md border bg-white p-6"
  >
    <h2
      class="mb-4 text-lg font-semibold"
    >
      Vehicle Specifications
    </h2>
    <div
      class="grid grid-cols-2 gap-3 text-sm"
    >
      <p>
        <strong>
          Brand/Model:
        </strong>
         Mercedes-Benz C200
      </p>
      <p>
        <strong>
          Color:
        </strong>
         Silver
      </p>
      <p>
        <strong>
          Year:
        </strong>
         2023
      </p>
      <p>
        <strong>
          Engine:
        </strong>
         2.0L Turbo
      </p>
      <p>
        <strong>
          Transmission:
        </strong>
         Automatic
      </p>
      <p>
        <strong>
          VIN:
        </strong>
         WDD2040**********
      </p>
      <p>
        <strong>
          Fuel Type:
        </strong>
         Petrol
      </p>
    </div>
    <button
      class="mt-4 rounded-md bg-blue-600 px-4 py-2 text-sm text-white hover:bg-blue-700"
      type="button"
    >
      Check Vehicle History
    </button>
  </div>
</div>
`;
